File: packages/create-react-native-library/templates/java-library-mixed/android/src/legacy/{%- project.name %}Module.java
package com.<%- project.package -%>;

import androidx.annotation.NonNull;
import com.facebook.react.bridge.Promise;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;

public class <%- project.name -%>Module extends ReactContextBaseJavaModule {
  public static final String NAME = <%- project.name -%>ModuleImpl.NAME;

  <%- project.name -%>Module(ReactApplicationContext context) {
    super(context);
  }

  @Override
  @NonNull
  public String getName() {
    return NAME;
  }

  // Example method
  // See https://reactnative.dev/docs/native-modules-android
  @ReactMethod
  public void multiply(double a, double b, Promise promise) {
    <%- project.name -%>ModuleImpl.multiply(a, b, promise);
  }
}

File: packages/create-react-native-library/templates/kotlin-library-mixed/android/src/legacy/{%- project.name %}Module.kt
package com.<%- project.package -%>;

import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod
import com.facebook.react.bridge.Promise

class <%- project.name -%>Module internal constructor(context: ReactApplicationContext?) :
  ReactContextBaseJavaModule(context) {

  override fun getName(): String {
    return NAME
  }

  // Example method
  // See https://reactnative.dev/docs/native-modules-android
  @ReactMethod
  fun multiply(a: Double, b: Double, promise: Promise) {
    <%- project.name -%>ModuleImpl.multiply(a, b, promise)
  }

  companion object {
    const val NAME = <%- project.name -%>ModuleImpl.NAME
  }
}
